plugin/storage/es/esCleaner.py

```diff
@@ -58,17 +62,29 @@ def main():
 
 
 def filter_main_indices(ilo, prefix):
-    ilo.filter_by_regex(kind='prefix', value=prefix + "jaeger")
+    ilo.filter_by_regex(kind='regex', value=prefix + "jaeger-(span|service|dependencies)-\d{4}-\d{2}-\d{2}")
     empty_list(ilo, "No indices to delete")
     # This excludes archive index as we use source='name'
     # source `creation_date` would include archive index
     ilo.filter_by_age(source='name', direction='older', timestring='%Y-%m-%d', unit='days', unit_count=int(sys.argv[1]))
 
 
-def filter_archive_indices(ilo, prefix):
-    # Remove only archive indices when aliases are used
+def filter_main_indices_rollover(ilo, prefix):
+    ilo.filter_by_regex(kind='regex', value=prefix + "jaeger-(span|service)-\d{6}")
+    empty_list(ilo, "No indices to delete")
+    # do not remove active write indices
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-span-write'], exclude=True)
+    empty_list(ilo, "No indices to delete")
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-service-write'], exclude=True)
+    empty_list(ilo, "No indices to delete")
+    ilo.filter_by_age(source='creation_date', direction='older', unit='days', unit_count=int(sys.argv[1]))
+
+
+def filter_archive_indices_rollover(ilo, prefix):
+    # Remove only rollover archive indices
     # Do not remove active write archive index
     ilo.filter_by_alias(aliases=[prefix + 'jaeger-span-archive-write'], exclude=True)
```

Comment:
Would it be better to filter by regex here aswell to be consistent with the change in `filter_main_indices_rollover`?

Reply:
unfortunatelly this is not `filter_by_regex` method

Reply:
Not sure what happened - the comment should have been associated with two lines down - so changing the `ilo.filter_by_alias(aliases=[prefix + 'jaeger-span-archive-read'])` to be `ilo.filter_by_regex(kind='regex', value=prefix + "jaeger-(span|service)-\d{6}")` ?

Reply:
Sorry,  I do not understand what you mean.

1. we use regex to find indices which should be removed
2. but we also need to exclude indices which are associated with write alias - to do that we use filter for an alias.

Reply:
Just mean that `filter_main_indices` and `filter_main_indices_rollover` are both using `filter_by_regex` to identify the indices to be removed - whereas `filter_archive_indices_rollover` uses `ilo.filter_by_alias(aliases=[prefix + 'jaeger-span-archive-read'])` - just suggesting that it would be better if `filter_archive_indices_rollover` also uses `filter_by_regex` for consistency.

Reply:
Yes, that sounds better https://github.com/jaegertracing/jaeger/pull/1693